Jryckman/update rulesetlogic

jryckman opened the pull request:
# Justification
There was a bug in the .targets file where it was always going to set NI.ruleset as the ruleset because it runs after the csproj defines a <CodeAnalysisRuleSet>. 

# Implementation
The update now checks to see if the ruleset is the default and only override it if it is.

# Testing
Hand testing with custom projects to verify it works

Commit:
Update NI.CSharp.Analyzers.targets

Update to the new DLLs we build and include in the package

Commit:
Merge remote-tracking branch 'upstream/main' into jryckman/update-analyzer-locations

Commit:
Update NI.CSharp.Analyzers.targets

Add code to .targest to properly honor CodeAnalysisRuleSet overrides

jryckman commented on src/AnalyzerConfiguration/NI.CSharp.Analyzers.targets:
```diff
@@ -14,10 +14,14 @@
-    <CodeAnalysisRuleSet Condition="'$(NI_IsTestProjectCondition)' == 'True'">$(NI_CodeAnalysisRuleSetDirectory)\NI.Tests.ruleset</CodeAnalysisRuleSet>
-    <CodeAnalysisRuleSet Condition="'$(NI_IsTestUtilitiesProjectCondition)' == 'True'">$(NI_CodeAnalysisRuleSetDirectory)\NI.TestUtilities.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSetDefined>False</CodeAnalysisRuleSetDefined>
+    <CodeAnalysisRuleSetDefined Condition="'$(CodeAnalysisRuleSet)' != 'MinimumRecommendedRules.ruleset'">True</CodeAnalysisRuleSetDefined>
```
MinimumRecommendedRules.ruleset is the default value that you get if you import Microsoft.CSharp.targets which every csproj needs in order to build with msbuild. I originally tried using empty string but it didn't work and this is why.

pakdev replied:
Thoughts on `<Condition="'$(CodeAnalysisRuleSet)' != '' and '$(CodeAnalysisRuleSet)' != 'MinimumRecommendedRules.ruleset'">`?

jryckman replied:
My main concern is if they try to blank out <CodeAnalysisRuleSet></CodeAnalysisRuleSet> because they didn't want one, then we'd overwrite it. I'm less worried about that because, well, they're installing this analyzer package so they want the rules. I'm torn, but I could see going with the '' check if you think it'll cover more cases.

Which case do you think is more likely to happen?

pakdev replied:
I assume if you don't have any ruleset defined the severities will come from each individual analyzer, and I doubt this is the intention of setting a blank ruleset.

pakdev commented on src/AnalyzerConfiguration/NI.CSharp.Analyzers.targets:
```diff
@@ -14,10 +14,14 @@
+    <CodeAnalysisRuleSetDefined>False</CodeAnalysisRuleSetDefined>
+    <CodeAnalysisRuleSetDefined Condition="'$(CodeAnalysisRuleSet)' != 'MinimumRecommendedRules.ruleset'">True</CodeAnalysisRuleSetDefined>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(CodeAnalysisRuleSetDefined)' == 'False'">
```
I think we put blank lines between PGs

Commit:
Update NI.CSharp.Analyzers.targets

Add newline between PropertyGroup and updated condition to also check for empty string
